deepchem/models/torch_models/tests/test_molgan_layers.py

```diff
@@ -235,3 +235,118 @@ def test_multigraph_convolution_layer_values():
         np.load('deepchem/models/tests/assets/molgan_multi_conv_layer_op.npy').
         astype(np.float32))
     assert torch.allclose(output, output_tensor, atol=1e-04)
+
+
+@pytest.mark.torch
+def test_graph_encoder_layer_shape():
+    from deepchem.models.torch_models.layers import MolGANEncoderLayer
+    vertices = 9
+    nodes = 5
+    edges = 5
+    first_convolution_unit = 128
+    second_convolution_unit = 64
+    aggregation_unit = 128
+    units = [(first_convolution_unit, second_convolution_unit),
+             aggregation_unit]
+
+    layer = MolGANEncoderLayer(units=units, edges=edges)
+    adjacency_tensor = torch.randn((1, vertices, vertices, edges))
+    node_tensor = torch.randn((1, vertices, nodes))
+    model = layer([adjacency_tensor, node_tensor])
+
+    assert model.shape == (1, aggregation_unit)
+    assert layer.graph_convolution_units == (first_convolution_unit,
+                                             second_convolution_unit)
+    assert layer.auxiliary_units == aggregation_unit
+    assert layer.activation == torch.tanh
+    assert layer.edges == 5
+    assert layer.dropout_rate == 0.0
+
+
+@pytest.mark.torch
+def test_graph_encoder_layer_values():
+    from deepchem.models.torch_models.layers import MolGANEncoderLayer
+    nodes = 5
+    edges = 5
+    first_convolution_unit = 128
+    second_convolution_unit = 64
+    aggregation_unit = 128
+    units = [(first_convolution_unit, second_convolution_unit),
+             aggregation_unit]
+
+    torch.manual_seed(21)
+    tf_weights = np.load(
+        'deepchem/models/tests/assets/molgan_encoder_layer_weights.npy',
+        allow_pickle=True).item()
+    torch_model_encoder = MolGANEncoderLayer(units=units,
+                                             nodes=nodes,
+                                             edges=edges,
+                                             name='layer1')
+
+    x = 12
```

Comment:
Why is x set to 12 here? In general we should avoid magic numbers without explanations in tests

Reply:
it was the starting number of the dense layers in the weights, have added a comment explaining the same